ingestors/maven.go

```diff
@@ -13,9 +13,13 @@ type MavenIngestor struct {
 }
 
 func NewMaven(feedName string) *MavenIngestor {
-	return &MavenIngestor{
+	maven := &MavenIngestor{
 		FeedName: feedName,
 	}
+	if maven.Name() == "" {
+		log.Fatalf("Unknown maven ingestor name: %s", feedName)
+	}
+	return maven
```

Comment:
not entirely sure about this style. How would you handle it? Sadly no const maps https://groups.google.com/g/golang-nuts/c/xUaDMciyB5Q?pli=1 as far as I can tell https://stackoverflow.com/questions/18342195/how-to-declare-a-constant-map-in-golang

Reply:
Hmm would something like this be nice?
```
type MavenRepository string

const(
    MavenCentral MavenRepository = "mavenCentral"
    Atlassian = "atlassian"
    Hortonworks = "hortonworks"
    SpringLibsRelease = "springLibsRelease"
)
```

Then we could initialize via
`func NewMaven(repository MavenRespository) *MavenIngestor` passing like `ingestors.MavenCentral`

Reply:
Yeah, bummer about const maps. I think the switch statement makes the most sense (altho not sure how that performs compared to a function that returns a map 🤷 )

Reply:
The main issue that we should tackle is exploding on init if something goofy is added. So whatever route we go, let's do that.

Reply:
> Hmm would something like this be nice?
> 
> ```
> type MavenRepository string
> 
> const(
>     MavenCentral MavenRepository = "mavenCentral"
>     Atlassian = "atlassian"
>     Hortonworks = "hortonworks"
>     SpringLibsRelease = "springLibsRelease"
> )
> ```
> 
> Then we could initialize via
> `func NewMaven(repository MavenRespository) *MavenIngestor` passing like `ingestors.MavenCentral`

Were you not into this idea? I think it's safer / easier to use.